src/bindings/python/src/openvino/runtime/utils/types.py

```diff
@@ -36,6 +36,8 @@
     (Type.u32, np.uint32),
     (Type.u64, np.uint64),
     (Type.bf16, np.uint16),
+    (Type.f8e4m3, np.float32),
```

Comment:
Will it be correct? 
If float32 will be used  then 4-bytes will be interpreted as values instead of one.

Maybe same approach should like with `bf16` to use `np.uint8` as there is no such types in numpy.
Add `f8e8m0` also?

Reply:
Does it needed in this PR? 
Mapping ov.f8e4m3/ov.f8e5m2 to np.float32 doesn't seem to be correct.
I would suggest to avoid extending this map without proper justification and verified choice of the destination type.

Reply:
I'll re-check, as I remember tests were failing in case this lines were missing. Maybe something has changed since first attempts.

Reply:
Answered https://github.com/openvinotoolkit/openvino/pull/25276#issuecomment-2213378862 , I cannot remove/change this code, otherwise it ruins tests.

Reply:
Mapping float8 to np.float32 is something used in onnx tests, but not correct globally.
 If needed for alignment with onnx backend test approach, I would suggest to add `get_onnx_test_dtype` helper and use it to replace `get_dtype` in the onnx test runtime here:
 https://github.com/openvinotoolkit/openvino/blob/6598aeb80e0bbc01698eb880a314dbb2e8b96ce8/src/frontends/onnx/tests/runtime.py#L138-L140
 
 with:
 
 ```
   def get_onnx_test_dtype(ov_type):
      is_ov_float8 = ov_type == Type.f8e4m3 or ov_type == Type.f8e5m2
      if is_ov_float8:
          return np.float32
      else:
          return get_dtype(ov_type)

  original_dtypes = [get_onnx_test_dtype(result.get_output_element_type(0)) for result in self.results]
 ```

Reply:
I don't think it is good idea. First of all, we have a bfloat16 (https://github.com/openvinotoolkit/openvino/blob/d55523cee0335815e5ca026c69580ec6200360a1/src/bindings/python/src/openvino/runtime/utils/types.py#L38) which is also has a "wrong" mapping.
Better even we keep wrong and useless float8 - int8 mapping (but I don't see any reason to do it) and re-map it later. When you remove original declaration in map, you loose consistency - you "overwriting" an absent mapping. Later, when someone will introduce "correct" (I still don't understand how you determine a correctness) mapping - it will cause a headache with understanding why onnx ignores a new declaration.
Also I don't understand what you are trying solve. It looks like: how to extend a map {'a'=>1, 'b' => 8} by {'c' => 3}? Hmmm, we can edit original map inplace... but we will introduce a new function in a totally different place which returns original map + {'c' => 3}. Why do we need this overcomplication? How does it make our code better/understandable?

Funny situation. I fully disagree with your suggestion, but I cannot merge the change without approval, so I obligated to do what I don't want to do, just to satisfy your requirements :)

Reply:
I'm not forcing anything, I was just looking for a solution to prevent casting f8 to np.float32 within this map, without blocking this PR, because as you mentioned uint8 was a problem as well. 
What I proposed has an impact on the onnx tests runtime only, and can be improved anytime.
Feel free to refactor this approach. The changes for the map can be discussed separately as well, with proper tests.
cc @AnastasiaKazantaeva